drivers/dma/dma_sedi.c

```diff
@@ -0,0 +1,533 @@
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)) {
+		LOG_ERR("dma reload failed for invalid args");
+		return -ENOTSUP;
+	}
+
+	int ret = 0;
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *config = &(data->dma_configs[channel]);
+	struct dma_block_config *block_config;
+
+	if ((config == NULL) || (config->head_block == NULL)) {
+		LOG_ERR("dma reload failed, no config found");
+		return -ENOTSUP;
+	}
+	block_config = config->head_block;
+
+	if ((config->block_count == 1) || (block_config->next_block == NULL)) {
+		block_config->source_address = src;
+		block_config->dest_address = dst;
+		block_config->block_size = size;
+	} else {
+		LOG_ERR("no reload support for multi-linkedlist mode");
+		return -ENOTSUP;
+	}
+	return ret;
+}
+
+static int dma_sedi_start(const struct device *dev, uint32_t channel)
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)) {
+		LOG_ERR("dma transferring failed for invalid args");
+		return -ENOTSUP;
+	}
+
+	int ret = -1;
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *config = &(data->dma_configs[channel]);
+	struct dma_block_config *block_config = config->head_block;
+	uint64_t src_addr, dst_addr;
+
+	if (config->block_count == 1) {
+		/* call sedi start function */
+		ret = dma_sedi_apply_single_config(info->peripheral_id,
+						   channel, config);
+		if (ret) {
+			goto ERR;
+		}
+		src_addr = block_config->source_address;
+		dst_addr = block_config->dest_address;
+
+		ret = sedi_dma_start_transfer(info->peripheral_id, channel,
+						src_addr, dst_addr, block_config->block_size);
+	} else {
+		LOG_ERR("MULTIPLE_BLOCK CONFIG is not set");
+		goto ERR;
+	}
+
+	if (ret != SEDI_DRIVER_OK) {
+		goto ERR;
+	}
+
+	return ret;
+
+ERR:
+	LOG_ERR("dma transfer failed");
+	return ret;
+}
+
+static int dma_sedi_stop(const struct device *dev, uint32_t channel)
+{
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *config = &(data->dma_configs[channel]);
+
+	LOG_DBG("stopping dma: %p, %d", dev, channel);
+	sedi_dma_abort_transfer(info->peripheral_id, channel);
+
+	free_blk_nodes(config);
+	return 0;
+}
+
+static const struct dma_driver_api dma_funcs = { .config = dma_sedi_chan_config,
+						 .start = dma_sedi_start,
+						 .stop = dma_sedi_stop,
+						 .reload = dma_sedi_reload,
+						 .get_status = NULL
+};
+
+static int dma_sedi_init(const struct device *dev)
+{
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	sedi_dma_t dma_dev = info->peripheral_id;
+	uint32_t irq_flag = 0;
+
+	switch (dma_dev) {
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+	case SEDI_DMA_0:
+#if DT_INST_IRQ_HAS_CELL(0, sense)
+		irq_flag = DT_INST_IRQ(0, sense);
+#endif
+		IRQ_CONNECT(DT_INST_IRQN(0),
+			    DT_INST_IRQ(0, priority),
+			    dma_isr, (void *)SEDI_DMA_0, irq_flag);
```

Comment:
This seems entirely unnecessary, the base address should be part of the DT spec, can't all this ifdefry be replaced with a single IRQ_CONNECT() where the IRQ number, priority, base address, and flag are obtained from the DT node properties?

Reply:
Follow other sedi code and change it.